nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -63,9 +64,11 @@ trap 'fail' 0
 
 
 # Print a greeting.
-echo
-echo "[1;32m<<< NixOS Stage 1 >>>[0m"
-echo
+if [ ! -z "$verbose" ]; then
```

Comment:
In general I would suggest to "factor" this pattern:

```shell
if [ -z $verbose ]; then
   echo useful strings
fi
```

It appears three times in this code, and certainly can be useful in other contexts. Maybe something like 

```shell
function printIfVerbose {
  if [ -z $verbose ]; then
    echo $@
  fi
}
```

Reply:
I would also have suggested something on those lines:

```diff
diff --git a/nixos/modules/system/boot/stage-1-init.sh b/nixos/modules/system/boot/stage-1-init.sh
index 1578c6cc34c..c80da688707 100644
--- a/nixos/modules/system/boot/stage-1-init.sh
+++ b/nixos/modules/system/boot/stage-1-init.sh
@@ -4,6 +4,12 @@ targetRoot=/mnt-root
 console=tty1
 verbose="@verbose@"
 
+info() {
+    if [[ -n "$verbose" ]]; then
+        echo "$@"
+    fi
+}
+
 extraUtils="@extraUtils@"
 export LD_LIBRARY_PATH=@extraUtils@/lib
 export PATH=@extraUtils@/bin
@@ -56,7 +62,7 @@ EOF
         echo "Rebooting..."
         reboot -f
     else
-        echo "Continuing..."
+        info "Continuing..."
     fi
 }
 
@@ -64,11 +70,9 @@ trap 'fail' 0
 
 
 # Print a greeting.
-if [ ! -z "$verbose" ]; then
-    echo
-    echo "[1;32m<<< NixOS Stage 1 >>>[0m"
-    echo
-fi
+info
+info "[1;32m<<< NixOS Stage 1 >>>[0m"
+info
 
 # Make several required directories.
 mkdir -p /etc/udev
@@ -213,18 +217,14 @@ ln -s @modulesClosure@/lib/modules /lib/modules
 ln -s @modulesClosure@/lib/firmware /lib/firmware
 echo @extraUtils@/bin/modprobe > /proc/sys/kernel/modprobe
 for i in @kernelModules@; do
-    if [ ! -z "$verbose" ]; then
-        echo "loading module $(basename $i)..."
-    fi
+    info "loading module $(basename $i)..."
     modprobe $i
 done
 
 
 # Create device nodes in /dev.
 @preDeviceCommands@
-if [ ! -z "$verbose" ]; then
-    echo "running udev..."
-fi
+info "running udev..."
 ln -sfn /proc/self/fd /dev/fd
 ln -sfn /proc/self/fd/0 /dev/stdin
 ln -sfn /proc/self/fd/1 /dev/stdout
@@ -242,9 +242,7 @@ udevadm settle
 # XXX: Use case usb->lvm will still fail, usb->luks->lvm is covered
 @preLVMCommands@
 
-if [ ! -z "$verbose" ]; then
-    echo "starting device mapper and LVM..."
-fi
+info "starting device mapper and LVM..."
 lvm vgchange -ay
 
 if test -n "$debug1devices"; then fail; fi
```

Reply:
Thanks to both of you for the review! This should take care of that